api/app/controllers/mno_enterprise/impersonate_controller.rb

```diff
@@ -10,7 +10,7 @@ class ImpersonateController < ApplicationController
     def create
       session[:impersonator_redirect_path] = params[:redirect_path].presence
       @user = MnoEnterprise::User.find(params[:user_id])
-      if @user.present?
+      if @user.present? && (!Settings.admin_panel || !Settings.admin_panel.impersonation.disabled)
```

Comment:
Instead of disabling it in the controller, you can disable it at the route level:

```ruby
unless Settings.try(:admin_panel).try(:impersonation).try(:disabled)
  get "/impersonate/user/:user_id", to: "impersonate#create", as: :impersonate_user
  get "/impersonate/revert", to: "impersonate#destroy", as: :revert_impersonate_user
end
```